core/src/main/java/feign/AsyncFeign.java

```diff
@@ -186,30 +186,32 @@ public AsyncBuilder<C> invocationHandlerFactory(InvocationHandlerFactory invocat
     }
 
     public AsyncFeign<C> build() {
-      super.enrich();
+      AsyncBuilder<C> enrichedBuilder = super.enrich();
```

Comment:
My major concern is that in the future, someone will access a field directly, forget the enrichBuilder prefix

What is,  we move the builder body to a new method, and the we invoke the "internalBuild()"  in the enriched instance?